### PR TITLE
Fix: clean up listeners on disconnection / diagnostic log only when it fails

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: packages/wallet-sdk/src/connection/WalletLinkConnection.ts [hunk 1]
```diff
@@ -227,11 +227,17 @@ export class WalletLinkConnection {
    * instance of WalletSDKConnection
    */
   public destroy(): void {
+    this.destroyed = true;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/connection/WalletLinkConnection.ts:230]: this should be set before calling `ws.disconnect()`

File: packages/wallet-sdk/src/connection/WalletLinkWebSocket.ts [hunk 2]
```diff
@@ -94,7 +94,11 @@ export class WalletLinkWebSocket {
       return;
     }
     this.clearWebSocket();
+
     this.connectionStateListener?.(ConnectionState.DISCONNECTED);
+    this.connectionStateListener = undefined;
+    this.incomingDataListener = undefined;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/connection/WalletLinkWebSocket.ts:100]: clean up after calling `this.connectionStateListener?.(ConnectionState.DISCONNECTED);`

File: packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts [hunk 3]
```diff
@@ -804,13 +804,20 @@ export class CoinbaseWalletProvider extends EventEmitter implements Web3Provider
...
-      sessionIdHash: this._relay ? Session.hash(this._relay.session.id) : undefined,
-    });
-
     const relay = await this.initializeRelay();
-    return relay.makeEthereumJSONRPCRequest(request, this.jsonRpcUrl);
+    return relay.makeEthereumJSONRPCRequest(request, this.jsonRpcUrl).catch((err) => {
+      if (
+        err.code === standardErrorCodes.rpc.methodNotFound ||
+        err.code === standardErrorCodes.rpc.methodNotSupported
+      ) {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts:812]: filter only these errors

**Review by hieronymus777 — APPROVED** [review]
Looks good

**Pull request merged**